### DISCUSSION TITLE
How to fetch data in Pinia?

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#17995] [Questions] [answered]
Hey there.
How i can pass data from `useFetch` to `Pinia` store? 

```
import { defineStore } from 'pinia';

export const useUserStore = defineStore('user', {
    state: () => ({
        user: null,
    }),

    actions: {
        async get() {
            const { data } = await useFetch('spa/auth/user');
        }
    }
});
```

I dont really understant how to pass `RefImpl` to the store of `Pinia`
Thanks.

**Comment by danielroe** [comment, marked as answer]
Rather than using `useFetch`, which is a composable meant for use in plugins or setup functions, use the `$fetch` method.

**Reply by k0mar12** [reply]
Yep, i guess you're right. I didnt get main goal of composable feature. thanks :)
